dev/com.ibm.ws.jpa.container.core/src/com/ibm/ws/jpa/management/JaxbUnmarshaller.java

```diff
@@ -278,8 +280,12 @@ public void startElement(String uri, String localName, String qName, Attributes
                 ivPersistence = new JaxbPersistence20(ivJPAPXml);
             } else if (JaxbPersistence21.SCHEMA_VERSION.equals(version)) {
                 ivPersistence = new JaxbPersistence21(ivJPAPXml);
-            } else {
+            } else if (JaxbPersistence22.SCHEMA_VERSION.equals(version)) {
                 ivPersistence = new JaxbPersistence22(ivJPAPXml);
+            } else if (JaxbPersistence30.SCHEMA_VERSION.equals(version)) {
```

Comment:
Actually, I had a couple of questions regarding this issue. One which I posted on #was-jakartaee-prototype, and the other regarding the forwards compatibility issue that you raised.

Right now, you can use the jpa-2.0 feature, and it'll read persistence.xml files that use later level persistence.xml schemas (of course, the APIs are not available.  But the persistence.xml schema has not changed since JPA 2.0). If I start enforcing the schema level to the spec level (that is, start throwing an error if a persistence.xml at the jpa 2.1 schema is presented, but the active feature is jpa 2.0), would that be a Zero Migration issue?  Can you comment, @NottyCode ?  Is it safe for me to implement this versioning restriction?

Reply:
As right now, if we have users who are doing this (probably unwittingly) in their applications, and we start enforcing this, then they would see their applications failing to start when they upgrade to the newer version of OL implementing this rule.

Reply:
I agree it would be a zero migration issue to start enforcing this for things in the past.  If they use the 2.2 persistence.xml in a 2.1 application, then sure, we need to continue supporting it.  However, going forward, we could prevent a user from using the 3.0 persistence.xml with an older JPA level.   Seems reasonable to allow then to use 2.x persistence.xml with the jpa 3.0 feature.